src/Tes.RunnerCLI/Commands/CommandLauncher.cs

```diff
@@ -31,47 +29,74 @@ private static void HandleResult(ProcessExecutionResult results, string command)
         {
             if (results.ExitCode != 0)
             {
-                throw new Exception(
-                    $"Task operation failed. Command: {command}. Exit Code: {results.ExitCode}{Environment.NewLine}Process Name: {results.ProcessName}");
+                throw new CommandExecutionException(results.ExitCode, $"Failed to execute command: {command}. Exit code returned by sub-process: {results.ExitCode}");
             }
         }
         /// <summary>
         /// Executes a transfer command as a sub-process. A transfer command is upload or download.
         /// </summary>
         /// <param name="command">Transfer command to execute</param>
         /// <param name="file">Node task definition file</param>
-        /// <param name="options">Transfer options</param>
-        /// <returns></returns>
+        /// <param name="options">Transfer options</param>        
```

Comment:
Consider removing whitespace at the end of this line

Reply:
x-ray vision!